### DISCUSSION TITLE
Reduce Animation Time

### DISCUSSION THREAD

**Discussion by hutber** [ihhub/fheroes2#8067] [General]
Is it possible to reduce one of the most annoying things about HoMM2 which is the animations times. Moving through portals, building things in towns, purchasing a hero etc.

I cannot fully articulate _why_ it happens, but my blood boils with these  delays. So much so its making me not want to play. of course this was in the original game. But do we really need this "bug" at least thats how I always perceived it.

Sorry to be negative :D As I love this game and being able to play it natively on linux is a dream come true!!!

also... I'd love to able to buy all creatures in a town and move them directly to the hero easily :D

**Comment by zenseii** [comment]
Hi, @hutber!

We have already done some work on being able to interrupt such animations/delays, like for the Puzzle Map and rightclicking to immediately start battle before the starting music finishes.

> also... I'd love to able to buy all creatures in a town and move them directly to the hero easily :D

You can already do this. Just open the Well and press the MAX button in the lower left corner, alternatively press M. To move all troops in the castle hit the down arrow key or up arrow key. We're still working on a nice UI way of doing this.

Also note that if there are no compatible or free slots in the castle garrison, the hero's army will be populated with the new troops if there is space.

Happy to hear that in general you're enjoying using the engine!

**Reply by hutber** [reply]
Dammm I need to read the controls properly! I assumed, as I have my alt key bound to be able to move windows with alt+click that I was missing some shortcuts, but these are much easier than I thought :D

Enjoying is an understatement :D Thank you very much!!
